File: java/28/ImplementstrStr.java
/**
 * @author: create by nadao
 * @version: v1.0
 * @description: 实现strStr()
 * @date:2019/11/19
 */
public class ImplementstrStr {
    public int strStr(String haystack, String needle) {
        if(needle == "") return 0;
        for (int i = 0; i <= haystack.length()-needle.length(); i++) {
            if(haystack.substring(i,i+needle.length()).equals(needle)){
                return i;
            }
        }
        return -1;
    }
}

File: python/28/Implement_strStr.py

class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        if not needle:
            return 0
        if not haystack:
            return -1
        for i in range(len(haystack)-len(needle)+1):
            j = 0
            if haystack[i] == needle[0]:
                k = i
                while j < len(needle) and haystack[k] == needle[j]:
                    k += 1
                    j += 1
                if j == len(needle):
                    return i
        return -1
